File: Design a Stack With Increment Operation.go
type CustomStack struct {
    N int
    Arr []int
}


func Constructor(maxSize int) CustomStack {
    return CustomStack{N: maxSize, Arr: make([]int, 0)}
}


func (this *CustomStack) Push(x int)  {
    if len(this.Arr) < this.N {
        this.Arr = append(this.Arr, x)
    }
}


func (this *CustomStack) Pop() int {
    if len(this.Arr) > 0 {
        val := this.Arr[len(this.Arr)-1]
        this.Arr = this.Arr[:len(this.Arr)-1]
        return val
    }
    return -1
}


func (this *CustomStack) Increment(k int, val int)  {
    for i := 0; i < min(k, len(this.Arr)); i++ {
        this.Arr[i] += val
    }
}


/**
 * Your CustomStack object will be instantiated and called as such:
 * obj := Constructor(maxSize);
 * obj.Push(x);
 * param_2 := obj.Pop();
 * obj.Increment(k,val);
 */

File: Design a Stack With Increment Operation.py
class CustomStack:

    def __init__(self, maxSize: int):
        self.n = maxSize
        self.arr = []

    def push(self, x: int) -> None:
        if len(self.arr) < self.n:
            self.arr.append(x)

    def pop(self) -> int:
        if len(self.arr) > 0:
            return self.arr.pop()
        return -1

    def increment(self, k: int, val: int) -> None:
        for i in range(min(k, len(self.arr))):
            self.arr[i] += val



# Your CustomStack object will be instantiated and called as such:
# obj = CustomStack(maxSize)
# obj.push(x)
# param_2 = obj.pop()
# obj.increment(k,val)
